### ISSUE TITLE
Outdated tabindex technique

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#526]
Copying over a public comment sent to public-comments-wcag20@w3.org from Chris O'Brien:

> Summary of Issue: Antiquated information - contradicts current best practices Comment (Including rationale for any proposed change):
> The current recommended techniques for tabindex contradict current best practices of avoiding positive tabindex values.
> 
> https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/H4#H4-resources
> 
> Have actually had clients challenge recommendations to avoid positive tabindex values because this technique (H4 Creating logical tab order...) still exists.
> 
> Proposed Change:
> Recommend retiring this technique

**Comment by Ryladog** [comment]
Chris talked to me about this at TPAC, so I asked him to file the issue

**Comment by alastc** [comment]
Thanks @Ryladog, good spot.

We can remove techniques, I notice H1 & H3 are not available any more. NB: They are removed across all versions of WCAG 2.x, not just the latest (2.1).

The technique does mention that positive values are navigated prior to "0" or default order items, but it isn't that obvious.

There is a sufficient technique (G59 make sure the source order is meaningful) which, given flexbox order & grid, is all we really have to work with these days. 

Does anyone want to keep this and update it? Otherwise I'd recommend removing it, I can't remember the last time this was a useful thing to do/recommend.

**Comment by Ryladog** [comment]
Thanks for looking into this Alastair.

As this is H4, which was the 4th HTML technique created, and H1 and H3 have
already been removed, maybe we should update it.

On Oct 24, 2018 5:11 PM, "Alastair Campbell" <notifications@github.com>
wrote:

Thanks @Ryladog <https://github.com/Ryladog>, good spot.

We can remove techniques, I notice H1 & H3 are not available any more. NB:
They are removed across all versions of WCAG 2.x, not just the latest (2.1).

The technique does mention that positive values are navigated prior to "0"
or default order items, but it isn't that obvious.

There is a sufficient technique (G59 make sure the source order is
meaningful) which, given flexbox order & grid, is all we really have to
work with these days.

Does anyone want to keep this and update it? Otherwise I'd recommend
removing it, I can't remember the last time this was a useful thing to
do/recommend.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/w3c/wcag/issues/526#issuecomment-432698652>, or mute
the thread

**Comment by alastc** [comment]
I don't think the numbering is a factor, we should keep it if there is a use case where this technique is useful & recommended.

**Comment by detlevhfischer** [comment]
Just wondering if there is a point in retaining positive tabindex for login pages and similar when focusable stuff comes before the login field, to cater for cases when authors want to refrain from setting focus to such fields when the page loads, but on the other hand want to ensure quick kb access.

**Comment by patrickhlauke** [comment]
the "i'm setting `tabindex` because i really want this to be the first thing that receives focus" angle, however, is one that arguably still results in general in an incorrect/illogical focus order, despite any well-meaning behind it. i'd say it's only ok in very specific situations (and even then, it's more of a "knowingly breaking the focus order for usability" reasons). having the technique there is potentially dangerous as it will signal that it's still ok to have a non-source-order focus sequence if the author wants something to be prominent/receive focus right away...unless we allow this sort of thing explicitly, it's probably dangerous territory.

removing the technique won't make those use cases "illegal", just that it won't have an example that shows it in action.

**Comment by detlevhfischer** [comment]
Had another look at H4.  I guess the examples suggest a use which goes beyond those few cases where it would be OK - and since it is a positive technique I agree that it should be retired.

**Comment by detlevhfischer** [comment]
@alastc You may want non-visual users to hear page title and such before sending them straight to the form?

**Comment by alastc** [comment]
@detlevhfischer Yes, there is probably a way of using .focus() after a short delay, but it’s fairly complex for a technique.

**Comment by jake-abma** [comment]
my 2 cents... also retire, causes only problems, and most of time, just as Patrick says, fixes the fast focus on a login for example but breaks illogical focus order if any other UIC is present on the screen.

**Comment by awkawk** [comment]
Agreed on WG call 5 Feb 2019 to retire H4.https://www.w3.org/2019/02/05-ag-minutes.html#item05

Leaving open until fully resolved.

**Comment by WayneEDick** [comment]
Hi All,
I ran into a necessary case when writing N2tPAssist
<http://nosetothepage.org/N2tPAssist/>. I inherited a lot of code from my
friend Tom Jewett. There was a position dependency in the code. It will be
really hard to fix, but I'll fix it on the next go around. Tom's UI was
dazzling.

I fixed  keyboard access by giving everything a tabindex, closing the page
off from other tabindex(s) with display none, and sending the last index
back to the start with an event listener. It was ugly coding, but
accessible.

Having worked at a site that always purchased code (a government agency). I
realized that many times people have to make do with what they have.

If you can produce a good keyboard experience with what you are given then
how you get there is not a failure of keyboard access.

Best, Wayne

**Comment by awkawk** [comment]
@WayneEDick And that is all still ok even if we remove the technique. The technique represents just one way to meet the SC and one that the WG doesn't want to promote, so it decided to retire it.

**Comment by alastc** [comment]
I snuck the [change](https://github.com/w3c/wcag/commit/9f455aeca4c7cf81b963d0a15bdeda63e3616a79) into master, closing now.
